src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceStreamingSource.java

```diff
@@ -154,6 +158,51 @@ private StructuredRecord getStructuredRecord(String jsonMessage) throws Connecti
     }
   }
 
+  private Object convertValue(Object value, Schema.Field field) {
+    if (value == null) {
+      return null;
+    }
+
+    Schema fieldSchema = field.getSchema();
+
+    if (fieldSchema.isNullable()) {
+      fieldSchema = fieldSchema.getNonNullable();
+    }
+
+    Schema.Type fieldSchemaType = fieldSchema.getType();
+    Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+
+    if (fieldSchema.getLogicalType() != null) {
+      String valueString = (String) value;
+      switch (logicalType) {
+        case DATE:
+          return Math.toIntExact(ChronoUnit.DAYS.between(Instant.EPOCH, Instant.parse(valueString)));
+        case TIMESTAMP_MICROS:
+          return TimeUnit.MILLISECONDS.toMicros(Instant.parse(valueString).toEpochMilli());
+        case TIME_MICROS:
+          return TimeUnit.NANOSECONDS.toMicros(LocalTime.parse(valueString).toNanoOfDay());
+        default:
+          throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'",
+                                                            field.getName(), logicalType.getToken()));
+      }
+    }
+
+    // Found a single field (Opportunity.Fiscal) which is documented as string and has a string type
```

Comment:
Let's also document this in the readme

Reply:
Do you think we should add this to a top level readme? This is a very specific small salesforce issue which is only specific to this plugin (not others) because of using streaming API. Not sure where to put it and where even such a small specific info should go into.

Reply:
I see - if it's only for streaming it's fine. Otherwise we should add it to the types map in the readme